### PR TITLE
HDDS-1786 : Datanodes takeSnapshot should delete previously created s…

### PR DESCRIPTION
**Pull request by avijayanhwx**
…napshots.

Right now, after after taking a new snapshot, the previous snapshot file is left in the raft log directory. When a new snapshot is taken, the previous snapshot should be deleted.

Fixed the issue in the takeSnapshot method. Manually tested on docker and added unit test.

### CODE REVIEW

**Comment by avijayanhwx** [comment]
@mukul1987 / @lokeshj1703 / @bharatviswa504 / @bshashikant Please review when you have a chance.

**Comment by avijayanhwx** [comment]
cc @swagle

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java [hunk 1]
```diff
@@ -256,6 +259,7 @@ public void persistContainerSet(OutputStream out) throws IOException {
   public long takeSnapshot() throws IOException {
     TermIndex ti = getLastAppliedTermIndex();
     long startTime = Time.monotonicNow();
+    SingleFileSnapshotInfo lastSnapshot = storage.findLatestSnapshot();
```
Comment (swagle) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java]: So this only deletes 1 previous snapshot, not all older than latest.

Comment (avijayanhwx) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java]: Yes

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineInt.java [hunk 2]
```diff
@@ -53,7 +53,7 @@
  * Tests the containerStateMachine failure handling.
  */
 
-public class TestContainerStateMachine {
+public class TestContainerStateMachineInt {
```
Comment (bharatviswa504) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineInt.java]: Why is this change done. Any reason for this?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java]: Got a question from above comments.
Just want to understand, every time when we take a snapshot, we are deleting old file means, only one snapshot file will be there. Is my understanding correct?

**Comment by bshashikant** [comment]
Thanks @avijayanhwx for working on this. In the patch , it seems at one point of time we will only have one snapshot in raft log directory. 
Can we have it as a configiurable property as to how many snapshots we want to retain at any point of time?

The default value can be 1. This will help in cases where the latest snapshot file gets corrupted/inaccessible but if we have older snapshots we can always roll back and start up from there. This will be more useful in debugging issues as well.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 107 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 3 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 28 | Maven dependency ordering for branch |
| +1 | mvninstall | 639 | trunk passed |
| +1 | compile | 398 | trunk passed |
| +1 | checkstyle | 75 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 939 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 182 | trunk passed |
| 0 | spotbugs | 502 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 748 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 34 | Maven dependency ordering for patch |
| +1 | mvninstall | 599 | the patch passed |
| +1 | compile | 404 | the patch passed |
| +1 | javac | 404 | the patch passed |
| +1 | checkstyle | 80 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 734 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 185 | the patch passed |
| +1 | findbugs | 766 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 370 | hadoop-hdds in the patch failed. |
| -1 | unit | 2318 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 44 | The patch does not generate ASF License warnings. |
| | | 8805 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.Test2WayCommitInRatis |
|   | hadoop.ozone.TestMiniOzoneCluster |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1163 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 5574c784c8f0 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / aebac6d |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/3/testReport/ |
| Max. process+thread count | 3999 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java (hunk 1 above)
Comment (avijayanhwx) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java]: Yes, but this will change.  I am planning to take up @bshashikant's suggestion.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineInt.java (hunk 2 above)
Comment (avijayanhwx) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestContainerStateMachineInt.java]: @bharatviswa504 I just changed this since I added a class called TestContainerStateMachine in correct package. So I added "int" to this class name to mark it as an integration test. Do you feel it is OK to have 2 classes with same name?

**Comment by avijayanhwx** [comment]
I will work on adding a configurable policy in Ratis so that Ozone Manager can also configure it as needed. After that is done, I will update this PR to use that policy.  cc @mukul1987

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 103 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 3 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for branch |
| +1 | mvninstall | 654 | trunk passed |
| +1 | compile | 373 | trunk passed |
| +1 | checkstyle | 74 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 970 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 192 | trunk passed |
| 0 | spotbugs | 487 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 739 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 34 | Maven dependency ordering for patch |
| +1 | mvninstall | 569 | the patch passed |
| +1 | compile | 390 | the patch passed |
| +1 | javac | 390 | the patch passed |
| +1 | checkstyle | 84 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 737 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 167 | the patch passed |
| +1 | findbugs | 724 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 369 | hadoop-hdds in the patch passed. |
| -1 | unit | 2083 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 8484 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.Test2WayCommitInRatis |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestOzoneClientRetriesOnException |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1163 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 8fba00b210ad 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / b94eba9 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/4/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/4/testReport/ |
| Max. process+thread count | 4989 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 46 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 623 | trunk passed |
| +1 | compile | 392 | trunk passed |
| +1 | checkstyle | 80 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 854 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 174 | trunk passed |
| 0 | spotbugs | 443 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 649 | trunk passed |
| -0 | patch | 491 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 552 | the patch passed |
| +1 | compile | 396 | the patch passed |
| +1 | javac | 396 | the patch passed |
| +1 | checkstyle | 85 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 654 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 175 | the patch passed |
| +1 | findbugs | 653 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 295 | hadoop-hdds in the patch passed. |
| -1 | unit | 1996 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 52 | The patch does not generate ASF License warnings. |
| | | 7864 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.TestSecureOzoneCluster |
|   | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.client.rpc.TestCloseContainerHandlingByClient |
|   | hadoop.ozone.scm.TestContainerSmallFile |
|   | hadoop.ozone.container.TestContainerReplication |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1163 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux d74699a82149 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 650c4ce |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/6/testReport/ |
| Max. process+thread count | 5268 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service U: hadoop-hdds/container-service |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 82 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 637 | trunk passed |
| +1 | compile | 372 | trunk passed |
| +1 | checkstyle | 76 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 954 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 170 | trunk passed |
| 0 | spotbugs | 436 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 638 | trunk passed |
| -0 | patch | 479 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 547 | the patch passed |
| +1 | compile | 377 | the patch passed |
| +1 | javac | 377 | the patch passed |
| +1 | checkstyle | 79 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 756 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 191 | the patch passed |
| +1 | findbugs | 657 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 314 | hadoop-hdds in the patch passed. |
| -1 | unit | 2269 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 8333 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.container.TestContainerReplication |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.TestSecureOzoneCluster |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1163 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 06da123aa662 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 650c4ce |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/7/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/7/testReport/ |
| Max. process+thread count | 4692 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service U: hadoop-hdds/container-service |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bshashikant** [comment]
Thanks @avijayanhwx for working on this. The changes look good. 
I think it would be better to move all configs related to RaftServer under the RaftServerConfig group but that's beyond the scope of this jira.

I would prefer to have a test in Ozone as well to verify the snapshot retention behaviour of Ratis so that, in case there are changes made in Ratis related to this, we should be able to catch this here in ozone.

A simple unit test where can change snapshot threshold to 1 entry and verify we have n snapshot files after n transactions in the raft log directory.

**Comment by avijayanhwx** [comment]
> Thanks @avijayanhwx for working on this. The changes look good.
> I think it would be better to move all configs related to RaftServer under the RaftServerConfig group but that's beyond the scope of this jira.
> 
> I would prefer to have a test in Ozone as well to verify the snapshot retention behaviour of Ratis so that, in case there are changes made in Ratis related to this, we should be able to catch this here in ozone.
> 
> A simple unit test where can change snapshot threshold to 1 entry and verify we have n snapshot files after n transactions in the raft log directory.

Added unit test.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 99 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 76 | Maven dependency ordering for branch |
| +1 | mvninstall | 678 | trunk passed |
| +1 | compile | 375 | trunk passed |
| +1 | checkstyle | 74 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 972 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 184 | trunk passed |
| 0 | spotbugs | 440 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 653 | trunk passed |
| -0 | patch | 483 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for patch |
| +1 | mvninstall | 546 | the patch passed |
| +1 | compile | 377 | the patch passed |
| +1 | javac | 377 | the patch passed |
| +1 | checkstyle | 77 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 728 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 169 | the patch passed |
| +1 | findbugs | 657 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 311 | hadoop-hdds in the patch passed. |
| -1 | unit | 2292 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 8517 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.scm.node.TestQueryNode |
|   | hadoop.ozone.TestSecureOzoneCluster |
|   | hadoop.ozone.container.TestContainerReplication |
|   | hadoop.ozone.client.rpc.TestBlockOutputStream |
|   | hadoop.ozone.client.rpc.TestContainerStateMachine |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1163 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 8d855eed86b8 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / dc9abd2 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/8/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/8/testReport/ |
| Max. process+thread count | 5342 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service hadoop-ozone/integration-test U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/8/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 573 | trunk passed |
| +1 | compile | 398 | trunk passed |
| +1 | checkstyle | 78 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 874 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 178 | trunk passed |
| 0 | spotbugs | 446 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 653 | trunk passed |
| -0 | patch | 500 | Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 40 | Maven dependency ordering for patch |
| +1 | mvninstall | 562 | the patch passed |
| +1 | compile | 400 | the patch passed |
| +1 | javac | 400 | the patch passed |
| +1 | checkstyle | 86 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 686 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 192 | the patch passed |
| +1 | findbugs | 789 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 314 | hadoop-hdds in the patch passed. |
| -1 | unit | 3129 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 9252 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerHandler |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerByPipeline |
|   | hadoop.ozone.client.rpc.TestDeleteWithSlowFollower |
|   | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.TestSecureOzoneCluster |
|   | hadoop.ozone.client.rpc.TestCommitWatcher |
|   | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.container.TestContainerReplication |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1163 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f632dd8cccfe 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / dc9abd2 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/9/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/9/testReport/ |
| Max. process+thread count | 4107 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service hadoop-ozone/integration-test U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1163/9/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bshashikant** [comment]
Thanks @avijayanhwx for working on the patch. The patch looks good to me. Can u check the compilation issue?

**Comment by avijayanhwx** [comment]
> Thanks @avijayanhwx for working on the patch. The patch looks good to me. Can u check the compilation issue?

@bshashikant I have fixed an edge case in the unit test.

**Comment by bshashikant** [comment]
Thanks @avijayanhwx for the contribution. I have committed this.
